assets/js/modules/analytics/datastore/report.js

```diff
@@ -274,6 +277,42 @@ const baseSelectors = {
 			return urlTitleMap;
 		}
 	),
+
+	/**
+	 * Determines whether the Analytics is still gathering data.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {boolean|undefined} Returns `true` if gathering data, otherwise `false`. Returns `undefined` while resolving.
+	 */
+	isGatheringData: createRegistrySelector( ( select ) => () => {
+		const { startDate, endDate } = select( CORE_USER ).getDateRangeDates();
+
+		const url = select( CORE_SITE ).getCurrentEntityURL();
+
+		const args = {
+			dimensions: [ 'ga:date' ],
+			metrics: [ { expression: 'ga:users' } ],
+			startDate,
+			endDate,
+		};
+
+		if ( url ) {
+			args.url = url;
+		}
+
+		const report = select( MODULES_ANALYTICS ).getReport( args );
+
+		if ( report === undefined ) {
+			return undefined;
+		}
+
+		if ( Array.isArray( report ) && report.length === 0 ) {
```

Comment:
I think the condition here should be if **not** an array **or** an empty array then return `true`.

Reply:
@eugene-manuilov really? Is my reading comprehension this bad or the IB is wrong? Can you have a look at the IB of this ticket?

Reply:
@kuasha420 Yes, IB says that it should only check whether it is an empty array, but we should also return `true` if the report data is `NULL` which is possible if I remember correctly.

Could you please check failed e2e and vrt checks?